src/migration/1713284688846-dataset.ts

```diff
@@ -0,0 +1,29 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class Dataset1713284688846 implements MigrationInterface {
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(
+            `
+          --Table Definition
+          CREATE TABLE IF NOT EXISTS "datasets"  (
+            "id" uuid NOT NULL DEFAULT gen_random_uuid(),
+            "internal_name" varchar(255) NOT NULL,
+            "creation_date" TIMESTAMP NOT NULL DEFAULT now(),
+            "created_by" varchar(255) NULL,
+            "last_modified" TIMESTAMP NOT NULL DEFAULT now(),
+            "modified_by" varchar(255) NULL,
+            "publish_date" TIMESTAMP NULL,
+            "published_by" varchar(255) null,
+            "live" BOOLEAN default false,
+            "approved_by" varchar(255) null,
+            "code" varchar(12) NULL,
+            CONSTRAINT "PK_dataset_id" PRIMARY KEY ("id")
+          );
```

Comment:
I can change the code so that we do this.  My concern is that we'll have multiple instances of the backend service running.  In theory, it should be near impossible for 2 separate machines to generate the same GUID at the same time but why risk it when the database is the ultimate source of truth and we'll only have one was my thinking.  But I am happy to iterate to make this change.  Which will untangle the logic a bit if we fail to write a file to the datalake as we won't have written to the database first.